ibm/data_source_ibm_tg_route_reports.go

```diff
@@ -0,0 +1,273 @@
+													Type:     schema.TypeBool,
+													Computed: true,
+												},
+												tgRouteReportConnectionBgpLocalPreference: {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												tgRouteReportConnectionBgpPrefix: {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									ID: {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									tgConnName: {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									tgRouteReportConnectionType: {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									tgRouteReportConnectionRoutes: {
+										Type:        schema.TypeList,
+										Description: "Collection of transit gateway connection's used routes",
+										Computed:    true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												tgRouteReportConnectionRoutePrefix: {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+
+						tgCreatedAt: {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						tgRouteReportOverlappingRoutes: {
+							Type:        schema.TypeList,
+							Description: "Collection of transit gateway overlapping routes",
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									tgRouteReportOverlappingRoutesDetail: {
+										Type:        schema.TypeList,
+										Description: "Collection of transit gateway overlapping route's details",
+										Computed:    true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												tgConnectionId: {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												tgRouteReportOverlappingPrefix: {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						tgStatus: {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						tgUpdatedAt: {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMTransitGatewayRouteReportsRead(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+
+	gatewayId := d.Get(tgGatewayId).(string)
+
+	listTransitGatewayRouteReportsOptionsModel := &transitgatewayapisv1.ListTransitGatewayRouteReportsOptions{}
+	listTransitGatewayRouteReportsOptionsModel.SetTransitGatewayID(gatewayId)
+	listTransitGatewayRouteReports, response, err := client.ListTransitGatewayRouteReports(listTransitGatewayRouteReportsOptionsModel)
```

Comment:
I think you need to add code for pagination to list all route reports

Reply:
Currently, there is no need for pagination. Only one route report per gateway is available at any time, as noted in the [customer docs](https://cloud.ibm.com/docs/transit-gateway?topic=transit-gateway-route-reports&interface=ui#generate-route-report-ui).

If this changes in the future, I will implement pagination.